### PR TITLE
Refactor/remove-kf-consolidate-manifest

### PR DESCRIPTION
**Pull request by Azher2Ali**
This PR refactors the manifest handling implementation in SCORe to consolidate support for a single manifest format, removing the KF-specific implementation in favor of a unified approach. Previously, SCORe provided separate implementations labeled 'icgc' and 'kf' for file download manifests.- #424 
 
- [x] Removed the java/bio/overture/score/client/manifest/kf package entirely.
- [x] Renamed IcgcManifestService to ManifestServiceImpl.
- [x]  Eliminated all references of @Profile("!kf") from classes in java/bio/overture/score/client/manifest.
- [x] Unified manifest download/upload implementations by refactoring Kf and icgc packages under manifest, creating common implementation classes.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -138,7 +138,7 @@ services:
       AWS_SECRET_ACCESS_KEY: minio123
       AWS_DEFAULT_REGION: us-east-1
     volumes:
-      - "./docker/object-storage-init/data/oicr.icgc.test/data:/score-data:ro"
+      - "./docker/object-storage-init/data/score.data/data:/score-data:ro"
```
Comment (joneubank) [inline docker-compose.yml]: Can you rename the directory this references too? This folder needs to match the volume referenced here.

https://github.com/overture-stack/score/tree/develop/docker/object-storage-init/data/oicr.icgc.test/data

**Review by joneubank — CHANGES_REQUESTED** [review]
All the manifest package changes are good.

Looks like you mixed in one commit here updating the docker buckets. That change needs an additional fix. If you want you can add in the fix to this PR and update the description to inlcude that change, or remove the commit that modified the docker-compose and move those changes to another PR.

**Comment by Azher2Ali** [comment]
The updated PR only contains changes related to Unifying the Manifests for KF and ICGC.

**Review by joneubank — DISMISSED** [review]

**Review by joneubank — DISMISSED** [review]
Repeating approval. Can you update the PR title to describe what is being done here, not autogenerated from the branch name?

**Comment by Azher2Ali** [comment]
> Repeating approval. Can you update the PR title to describe what is being done here, not autogenerated from the branch name?

Earlier this PR also included changes other than unifying manifest, which later was partioned into a seperate ticket. Now this PR only holds changes to eliminating the KF profiles and refacorting manifests.

**Review by joneubank — DISMISSED** [review]
Please update the PR title to describe the changes being made instead of being the text autogenerated from the branch name.

**Review by leoraba — DISMISSED** [review]
LGTM

**Review by joneubank — DISMISSED** [review]
Reapproving after stale review was dismissed.

**Comment by joneubank** [comment]
I don't know why the approvals keep getting dismissed here. @Azher2Ali do you need a hand getting this merged?

**Comment by Azher2Ali** [comment]
I just found out that the issue is with rebasing. Let me just take a final look, if that's not helping will surely reach out.

**Comment by Azher2Ali** [comment]
Closing this PR, due to rebasing conflicts

**Pull request closed without merging**